pcdsdevices/slits.py

```diff
@@ -301,6 +203,221 @@ def _aperture_changed(self, *args, **kwargs):
+
+    # Local PVs
+    blocked = Cpt(EpicsSignalRO, ':BLOCKED', kind='omitted')
+    open_cmd = Cpt(EpicsSignal, ':OPEN', kind='omitted')
+    close_cmd = Cpt(EpicsSignal, ':CLOSE', kind='omitted')
+    block_cmd = Cpt(EpicsSignal, ':BLOCK', kind='omitted')
+
+    def open(self):
+        """Uses the built-in 'OPEN' record to move open the aperture."""
+        self.open_cmd.put(1)
+
+    def close(self):
+        """Close the slits to have an aperture of 0mm on each side."""
+        self.close_cmd.put(1)
+
+    def block(self):
+        """Overlap the slits to block the beam."""
+        self.block_cmd.put(1)
+
+
+class Slits(LusiSlits):
+    # Should Probably Deprecate this name
+    pass
+
+
+class BeckhoffSlitPositioner(BadSlitPositionerBase):
+    """
+    Abstraction of a Slit axis from LCLS-II.
+
+    This class needs a BeckhoffSlits parent to function properly.
+    """
+
+    readback = FCpt(PytmcSignal, BadSlitPositionerBase.readback.suffix,
+                    io='i', auto_monitor=True, kind='normal')
+    setpoint = FCpt(PytmcSignal, BadSlitPositionerBase.setpoint.suffix,
+                    io='io', auto_monitor=True, kind='normal')
+    done = Cpt(Signal, kind='omitted')
+    actuate = Cpt(Signal, kind='omitted')
+
+    @actuate.sub_value
+    def _execute_move(self, *args, value, **kwargs):
+        if value == 1:
+            self.parent.exec_queue.put(1)
+
+    @done.sub_value
+    def _reset_actuate(self, *args, value, old_value, **kwargs):
+        if value == 1 and old_value == 0:
+            self.actuate.put(0)
+
+
+class BeckhoffSlits(SlitsBase):
+    # Base class overrides
+    xwidth = Cpt(BeckhoffSlitPositioner, '', slit_type='XWIDTH', kind='hinted')
+    ywidth = Cpt(BeckhoffSlitPositioner, '', slit_type='YWIDTH', kind='hinted')
+    xcenter = Cpt(BeckhoffSlitPositioner, '', slit_type='XCENTER',
+                  kind='normal')
+    ycenter = Cpt(BeckhoffSlitPositioner, '', slit_type='YCENTER',
+                  kind='normal')
+
+    # Slit state commands
+    exec_queue = Cpt(Signal, kind='omitted')
+    exec_move = Cpt(PytmcSignal, ':GO', io='io', kind='omitted')
+
+    # Slit calculated move dmov
+    done_all = Cpt(Signal, kind='omitted')
+    done_top = Cpt(PytmcSignal, ':TOP:DMOV', io='i', kind='omitted')
+    done_bottom = Cpt(PytmcSignal, ':BOTTOM:DMOV', io='i', kind='omitted')
+    done_north = Cpt(PytmcSignal, ':NORTH:DMOV', io='i', kind='omitted')
+    done_south = Cpt(PytmcSignal, ':SOUTH:DMOV', io='i', kind='omitted')
+
+    # Raw motors
+    top = Cpt(BeckhoffAxis, ':MMS:TOP', kind='normal')
+    bottom = Cpt(BeckhoffAxis, ':MMS:BOTTOM', kind='normal')
+    north = Cpt(BeckhoffAxis, ':MMS:NORTH', kind='normal')
+    south = Cpt(BeckhoffAxis, ':MMS:SOUTH', kind='normal')
+
+    @exec_queue.sub_value
+    def _exec_handler(self, *args, value, old_value, **kwargs):
+        """Wait just a moment to queue up move requests."""
+        if value == 1 and old_value == 0:
+            time.sleep(0.2)
```

Comment:
Blocking here isn't great as it holds up all callbacks for 200ms and may cause timeouts.

I suppose a `Timer` that adds `exec_move.put(1)` to the ophyd callback queue would be good

Reply:
I agree, I couldn't come up with an alternative while going through the first time. I will fix.